bonds/src/utils/Price.ts

```diff
@@ -66,8 +67,8 @@ export function getKLIMAMCO2Rate(): BigDecimal {
 }
 
 //(slp_treasury/slp_supply)*(2*sqrt(lp_dai * lp_ohm))
-export function getDiscountedPairCO2(lp_amount: BigInt, pair_address: string): BigDecimal {
-    let pair = UniswapV2Pair.bind(Address.fromString(pair_address))
+export function getDiscountedPairCO2(lp_amount: BigInt, pair_address: Address): BigDecimal {
+    let pair = UniswapV2Pair.bind(pair_address)
 
     let total_lp = pair.totalSupply()
     let lp_token_1 = toDecimal(pair.getReserves().value0, 9)
```

Comment:
Would it be possible to retrieve the reserve tokens and pull the correct decimals from the result? We can't guarantee that KLIMA will be first and reserve second on the pair.

Reply:
Did something like this:

`//(slp_treasury/slp_supply)*(2*sqrt(lp_dai * lp_ohm))
export function getDiscountedPairCO2(lp_amount: BigInt, pair_address: Address, pairToken1: IToken, pairToken2: IToken): BigDecimal {
    let pair = UniswapV2Pair.bind(pair_address)

    let total_lpCall = pair.try_totalSupply()
    if (total_lpCall.reverted) {
        return BigDecimal.zero()
    }
    
    let total_lp = total_lpCall.value
    let lp_token_1 = toDecimal(pair.getReserves().value0, pairToken1.getDecimals())
    let lp_token_2 = toDecimal(pair.getReserves().value1, pairToken2.getDecimals())
    let kLast = lp_token_1.times(lp_token_2).truncate(0).digits

    let part1 = toDecimal(lp_amount, 18).div(toDecimal(total_lp, 18))
    let two = BigInt.fromI32(2)

    let sqrt = kLast.sqrt();
    let part2 = toDecimal(two.times(sqrt), 0)
    let result = part1.times(part2)
    return result
}`